### PR TITLE
[kube-prometheus-stack] Allow annotations on the admissionWebhook Jobs

### LINKED ISSUE #2516: [kube-prometheus-stack] validation and mutation webhook failurePolicy OutOfSync
### Describe the bug a clear and concise description of what the bug is.

When install `prometheus-kube-stack` with ArgoCD, it become OutOfSync, cause `admissionWebhooks.failurePolicy` and `admissionWebhooks.patch.enabled` produce different values.
In the webhooks configuration we have [following condition](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/mutatingWebhookConfiguration.yaml#L16-L19):
```
{{- if .Values.prometheusOperator.admissionWebhooks.patch.enabled  }}
    failurePolicy: Ignore
    {{- else }}
    failurePolicy: {{ .Values.prometheusOperator.admissionWebhooks.failurePolicy }}
    {{- end }}
```
By this one we should have failure policy `Ignore`. But, as we have `prometheusOperator.admissionWebhooks.patch.enabled : true` we are running [patch job](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job-patch/job-patchWebhook.yaml#L45), which change policy to `Fail` and leave ArgoCD OutofSync.

Screenshot 2022-10-03 at 12 09 06

### What's your helm version?

3.8.0

### What's your kubectl version?

1.22.2

### Which chart?

kube-prometheus-stack

### What's the chart version?

40.2.0

### What happened?

ArgoCD leave OutOfSync state after sync finish.leaves

### What you expected to happen?

ArgoCD leave in Sync state.leaves

### How to reproduce it?

Install latest chart with ArgoCD.

### Enter the changed values of values.yaml?

_No response_

### Enter the command that you execute and failing/misfunctioning.

ArgoCD version: 2.4.12

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by marcofranssen**
#### What this PR does / why we need it

When deploying via ArgoCD the ArgoCD app will be in a synchronization loop due to the kube-prometheus-stack jobs being removed due the `ttlAfterFinished: 0`. Using annotations we can inform ArgoCD not to recreate the Jobs in an infinite loop.

Currently the jobs can't be annotated and therefore we need this PR to allow the deployment of `kube-prometheus-stack` using ArgoCD.

#### Which issue this PR fixes

- fixes #2516

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

@andrewgkew @desaintmartin @gianrubio @gkarthiks @scottrigby @Xtigyro

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/values.yaml [hunk 1]
```diff
@@ -1611,6 +1611,10 @@ prometheusOperator:
     ## On chart upgrades (or if the secret exists) the cert will not be re-generated. You can use this to provide your own
     ## certs ahead of time if you wish.
     ##
+    annotations: {}
+    # annotations:
```
[the remark this answers is not in the document]
Comment (marcofranssen) [reply charts/kube-prometheus-stack/values.yaml:1616]: What is the difference here? I don't see what changed here.

File: charts/kube-prometheus-stack/Chart.yaml [hunk 2]
```diff
@@ -22,7 +22,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 40.5.0
+version: 40.5.1
```
[the remark this answers is not in the document]
Comment (marcofranssen) [reply charts/kube-prometheus-stack/Chart.yaml]: Next minor version number is `41.2.0`. I have bumped it there.

**Comment by marcofranssen** [comment]
@monotek Thanks for the feedback. Processed your feedback, though I did bump the version to `41.2.0` as that is the next patch release.

File: charts/kube-prometheus-stack/Chart.yaml [hunk 3]
```diff
@@ -22,7 +22,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 41.1.0
+version: 41.2.0
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:25]:
```suggestion
version: 41.3.0
```

**Review by monotek — CHANGES_REQUESTED** [review]
41.2.0 was just merged.

**Pull request merged**